hadoop/Dockerfile

```diff
@@ -153,5 +154,20 @@ ENV ASYNC_PROFILER_HOME=/stackable/async-profiler
 ENV HADOOP_YARN_HOME=/stackable/hadoop
 ENV HADOOP_MAPRED_HOME=/stackable/hadoop
 
+# Remove unneeded binaries:
+#  - source code
+#  - mapreduce/yarn binaries that were built as cross-project dependencies
+#  - minicluster (only used for testing) and test .jars
+#  - json-io: this is a transitive dependency pulled in by cedarsoft/java-utils/json-io and is excluded in 3.4.0. See CVE-2023-34610.
+RUN rm -rf /stackable/hadoop/share/hadoop/common/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/hdfs/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/json-io-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-mapreduce-client-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-yarn-server*.jar && \
+  find . -name hadoop-minicluster-*.jar -type f -delete && \
+  find . -name hadoop-client-minicluster-*.jar -type f -delete && \
+  find . -name hadoop-*tests.jar -type f -delete
```

Comment:
There is the shellcheck warning which actually might make sense here to look at.

Reply:
See 878cc24.